### PR TITLE
Add docs about auto-restart and checks

### PR DESCRIPTION
**Pull request by benhoyt**
These are updates to the [How to interact with Pebble](https://juju.is/docs/sdk/pebble) document ([Discourse original here](https://discourse.charmhub.io/t/how-to-interact-with-pebble/4554)) for the new Pebble auto-restart and health checks features.

Implementation for these features is in https://github.com/canonical/pebble/pull/85 and https://github.com/canonical/pebble/pull/86.

### CODE REVIEW

File: sdk/pebble.md [hunk 1]
```diff
@@ -132,6 +132,111 @@ class MyCharm(CharmBase):
...
+The `backoff-factor` must be greater than or equal to 1.0. If the factor is set to 1.0, `next` will equal `current`, so the delay will remain constant.
+
+Just before delaying, a small random time jitter of 0-10% of the delay is added (the current delay is not updated). For example, if the current delay value is 2 seconds, the actual delay will be between 2.0 and 2.2 seconds.
+
+A charm can find the total number of times Pebble has restarted a service using the `restarts` field in the [`ServiceInfo`](https://ops.readthedocs.io/en/latest/#ops.pebble.ServiceInfo) object returned by the [`Container.get_service`](https://ops.readthedocs.io/en/latest/#ops.model.Container.get_service) method.
+
+
+<h2 id="heading--health-checks">Health checks</h2>
+
+From Juju version TODO:TBD, Pebble supports adding custom health checks: first, to allow Pebble itself to restart services when certain checks fail, and second, to allow Kubernetes to restart containers when specified checks fail.
```
Comment (tmihoc) [inline sdk/pebble.md:174]: Need to add version.

**Review by tmihoc — COMMENTED** [review]
Main question: For layer specification, I see a Github link. Is that the only place where we document that currently? Other than that, the Juju versions are missing, but I know you were planning to add them later, and there are a couple of things I'd rephrase, and but I can implement that directly on Discourse, once this is published, and I can also make sure to properly anchor all the links (just the h tag unfortunately doesn't do the job).

**Comment by benhoyt** [comment]
> For layer specification, I see a Github link. Is that the only place where we document that currently?

@tmihoc Yes, that's right -- it's the reference doc for the layer format. We *could* copy it over to Discourse / SDK docs, I suppose, though it'd be a bit tedious to update in two places. What do you think?

Yes, I was planning to add the Juju versions once the code is merged and we know what version of Juju it'll be going in.

**Comment by tmihoc** [comment]
@benhoyt Wondering about this. In principle I want all documentation to be in the same place. That should be an authoritative place, like the actual product website, for example, here, the SDK docs. However, there are of course all these other places where the project is actually hosted and we should, of course, describe the project there too. For that reason I think we won't be able to avoid some amount of duplication. But what we can do is try to keep this duplication minimal. For example, the Github readme should contain no more than one paragraph, maybe even copy-pasted from the SDK docs, and then simply say: For more, visit the SDK docs. Or something like this. I'll think about it more.
